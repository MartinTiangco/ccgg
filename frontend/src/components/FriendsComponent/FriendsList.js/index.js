import React from "react";
import { makeStyles } from "@material-ui/core/styles";
import {
  Card,
  Grid,
  TextField,
  Typography,
  Button,
  Tooltip,
} from "@material-ui/core";
import ErrorOutlineIcon from "@material-ui/icons/ErrorOutline";
import AddIcon from "@material-ui/icons/Add";
import missingIcon from "../../../images/icons/missing.png";

const useStyles = makeStyles((theme) => ({
  summonerName: {
    color: "#A6ACAF",
  },
  iconImg: {
    maxHeight: "2em",
  },
  add: {
    backgroundColor: "white",
    color: theme.palette.primary.main,
    maxHeight: "6em",
    padding: "5px",
    margin: "5px",
  },
  noBorder: {
    border: "none",
  },
  textFieldColor: {
    color: theme.palette.primary.main,
  },
  friendRow: {
    color: theme.palette.text.secondary,
    padding: theme.spacing(2),
    margin: "5px",
    "&:hover": {
      filter: "brightness(110%)",
    },
    minHeight: "6em",
  },
  formControl: {
    margin: theme.spacing(1),
    minWidth: 120,
  },
  scroll: {
    overflowY: "scroll",
    maxHeight: "67vh",
    "&::-webkit-scrollbar": {
      width: 8,
    },
    "&::-webkit-scrollbar-track": {
      boxShadow: `inset 0 0 6px rgba(0, 0, 0, 0.3)`,
      borderRadius: "3px",
    },
    "&::-webkit-scrollbar-thumb": {
      backgroundColor: "white",
      borderRadius: "3px",
    },
  },
}));

const FriendsList = ({ friends, handlePingFriend }) => {
  const classes = useStyles();

  // eslint-disable-next-line
  const [inputText, setInputText] = React.useState("");
  const [invalidFriend, setInvalidFriend] = React.useState(false);

  const noFriendsComponent = (
    <Grid
      item
      container
      direction="column"
      justify="center"
      alignItems="center"
    >
      <ErrorOutlineIcon />
      No friends found!
    </Grid>
  );

  const FriendsComponent = friends.map((friend) => {
    const { username, summonerName } = friend;

    return (
      <Card key={username} className={classes.friendRow}>
        <Grid container justify="space-between">
          <Grid container item direction="column" xs={11}>
            <Grid container item xs={4}>
              <Typography variant="h5">{username}</Typography>
            </Grid>
            <Grid container item xs={7}>
              <Typography variant="body2" className={classes.summonerName}>
                Summoner: {summonerName}
              </Typography>
            </Grid>
          </Grid>
          <Grid container item xs={1}>
            <Tooltip title="Ping Friend">
              <Button
                color="secondary"
                onClick={() => {
                  handlePingFriend(summonerName);
                }}
              >
                <img
                  src={missingIcon}
                  alt="Ping Icon"
                  className={classes.iconImg}
                />
              </Button>
            </Tooltip>
          </Grid>
        </Grid>
      </Card>
    );
  });

  const currentFriendsComponent =
    friends.length === 0 ? noFriendsComponent : FriendsComponent;

  const handleSubmit = () => {
    setInvalidFriend(!invalidFriend);
    // TODO: CALL BACKEND TO ADD FRIEND
  };

  const handleKeyPress = (event) => {
    if (event.key === "Enter") {
      // TODO CALL BACKEND
    }
  };

  return (
    <Grid container direction="column">
      <Card className={classes.add}>
        <Grid container item direction="row" alignItems="center" xs={12}>
          <Grid container item xs={11}>
            <TextField
              id="outlined-basic"
              variant="outlined"
              fullWidth
              label="Enter username..."
              InputLabelProps={{
                className: classes.textFieldColor,
              }}
              InputProps={{
                className: classes.textFieldColor,
                classes: { notchedOutline: classes.noBorder },
              }}
              onInput={(e) => setInputText(e.target.value)}
              onKeyPress={handleKeyPress}
              error={invalidFriend}
              helperText={invalidFriend ? "Invalid username!" : ""}
              autoComplete="off"
            />
          </Grid>
          <Grid container item xs={1} direction="column">
            <Tooltip title="Add Friend">
              <Button color="primary" onClick={handleSubmit}>
                <AddIcon />
              </Button>
            </Tooltip>
          </Grid>
        </Grid>
      </Card>
      <Grid
        container
        item
        direction="column"
        wrap="nowrap"
        xs={12}
        className={classes.scroll}
      >
        {currentFriendsComponent}
      </Grid>
    </Grid>
  );
};

export default FriendsList;
