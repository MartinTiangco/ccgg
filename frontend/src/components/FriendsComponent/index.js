import React from "react";
import { makeStyles } from "@material-ui/core/styles";
import { Card, Grid, Typography } from "@material-ui/core";
import { useAuth0 } from "@auth0/auth0-react";
import FriendsList from "./FriendsList.js";

// remove this/change this once integrated with backend
const testFriends = [
  { username: "bobbobobobobobobob", summonerName: "bob1111112222" },
  { username: "bob1", summonerName: "bob2" },
  { username: "bob3", summonerName: "bob4" },
  { username: "bobbobobobobobobobobobob", summonerName: "bob111111" },
  { username: "bob", summonerName: "bob2" },
  { username: "bob3ef", summonerName: "bob4" },
  { username: "bobboefboboboboboobobob", summonerName: "bob111111" },
  { username: "bobef1", summonerName: "bob2" },
  { username: "bo", summonerName: "bob4" },
  { username: "bobboboboboboefbobobobobob", summonerName: "bob111111" },
  { username: "befob1", summonerName: "bob2" },
  { username: "bob3effe", summonerName: "bob4" },
  { username: "bobboboboboboefbobobob", summonerName: "bob111111" },
  { username: "bob1fe", summonerName: "bob2" },
  { username: "boefeb3", summonerName: "bob4" },
];

const useStyles = makeStyles((theme) => ({
  root: {
    [theme.breakpoints.down("sm")]: {
      flexDirection: "column",
    },
    [theme.breakpoints.up("sm")]: {
      flexDirection: "row",
    },
    margin: "20px",
    overflow: "hidden",
  },
  innerGrid: {
    margin: "0 auto",
  },
  paper: {
    padding: theme.spacing(8),
    color: theme.palette.text.secondary,
  },
  card: {
    padding: theme.spacing(2),
    color: theme.palette.text.secondary,
  },
}));

const FriendsComponent = () => {
  const classes = useStyles();
  const { user, isLoading } = useAuth0();

  // If the auth0 context is still loading, return blank page
  if (isLoading) {
    return <div />;
  }

  return (
    <Grid container className={classes.root}>
      <Grid
        container
        item
        className={classes.innerGrid}
        direction="column"
        xs={10}
        sm={7}
        md={8}
        spacing={2}
      >
        <Grid
          container
          item
          direction="row"
          justify="space-between"
          alignItems="center"
        >
          <Typography variant="h4">Friends List</Typography>
          <Typography variant="h4">
            Username: {user["https://cc.gg/user_metadata"].summonerName}
          </Typography>
        </Grid>
        <Grid item>
          <Card className={classes.card} raised>
            <FriendsList friends={testFriends} />
          </Card>
        </Grid>
      </Grid>
    </Grid>
  );
};

export default FriendsComponent;
