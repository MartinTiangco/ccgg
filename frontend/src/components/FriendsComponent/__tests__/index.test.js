import renderer from "react-test-renderer";
import React from "react";
import FriendsComponent from "../index";

it("snapshot test of FriendsComponent", () => {
  const tree = renderer.create(<FriendsComponent />).toJSON();
  expect(tree).toMatchSnapshot();
});
