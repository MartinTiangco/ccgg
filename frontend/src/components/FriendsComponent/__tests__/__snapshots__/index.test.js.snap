// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test of FriendsComponent 1`] = `<div />`;
